weather/www/js/services.js

```diff
@@ -0,0 +1,214 @@
+/* Notes on forecast.io's API:
+ *  - daily.data[0] is today
+ * By: Cody Davey & Travis Pullen
+ */
+
+/* weatherStore holds data for weatherData service.
+ */
+weatherApp.factory('weatherStore', function() {
+    var weatherStore = {
+        current : {}
+    };
+
+    return weatherStore;
+});
+
+/* weatherData service gets weather data (current, forecasts, and
+ * historical), processes it as necessary, and provides it to controllers.
+ */
+weatherApp.service('weatherData', ['$q', '$resource', '$http',
+                                   'FORECASTIO_KEY', 'weatherStore',
+    function($q, $resource, $http, FORECASTIO_KEY, weatherStore) {
+        this.getCurrentWeather = function(lat, lng) {
+            var url = 'https://api.forecast.io/forecast/' +
+                FORECASTIO_KEY + "/" + lat + ',' + lng;
+
+            // JSONP is only needed for "ionic serve".
+            // Simpler $http.get(url) works on devices.
+            return $http.jsonp(url + '?callback=JSON_CALLBACK').then(
+                function success(resp) {
+                    weatherStore.current = resp.data;
+                    console.log('GOT CURRENT');
+                    console.dir(weatherStore.current);
+                },
+                function failure(error) {
+                    alert('Unable to get current conditions');
+                    console.error(error);
+                });
+        };
+
+        this.getLocation = function() {
```

Comment:
This is a good feature but if you give weather for current position then the UI should somewhere mention that the weather is for the current location